### PR TITLE
Add reset api option on change password

### PR DESCRIPTION
**Pull request by spk**
ref https://github.com/rubygems/rubygems.org/issues/1951

Screenshot-2019-06-10-121325

### CODE REVIEW

File: app/controllers/passwords_controller.rb [hunk 1]
```diff
@@ -9,6 +9,20 @@ def edit
     end
   end
 
+  def update
+    @user = find_user_for_update
+
+    if @user.update_password password_reset_params
+      @user.reset_api_key! if params[:password_reset] && params[:password_reset][:reset_api_key].in?([true, '1', 1])
```
Comment (sonalkr132) [inline app/controllers/passwords_controller.rb]: Can we please use strong param here?

File: app/views/passwords/edit.html.erb [hunk 2]
```diff
@@ -8,6 +8,10 @@
     <%= form.label :password, "Password", :class => 'form__label' %>
     <%= form.password_field :password, :class => 'form__input' %>
   </div>
+  <div class="form__checkbox">
+    <%= form.check_box :reset_api_key, :class => 'form__checkbox__input' %>
```
Comment (sonalkr132) [inline app/views/passwords/edit.html.erb]: Let's pin form value here and check for just that in the controller.
```
form.check_box :reset_api_key, { :class => 'form__checkbox__input' } , 'true', 'false'
```

File: app/controllers/passwords_controller.rb (hunk 1 above)
Comment (spk) [reply app/controllers/passwords_controller.rb]: Yep fixed in https://github.com/spk/rubygems.org/commit/b529b055f583973024f71459e6ef350a21fea1c1

File: app/views/passwords/edit.html.erb (hunk 2 above)
Comment (spk) [reply app/views/passwords/edit.html.erb]: Done in https://github.com/spk/rubygems.org/commit/2007dc6a51be9aa2f0805aa868333655385171a8

**Pull request merged**